src/sentry/api/endpoints/notifications/notification_actions_details.py

```diff
@@ -34,7 +35,15 @@ class NotificationActionsDetailsEndpoint(OrganizationEndpoint):
     def convert_args(self, request: Request, action_id: int, *args, **kwargs):
         parsed_args, parsed_kwargs = super().convert_args(request, *args, **kwargs)
         organization = parsed_kwargs["organization"]
+        # projects where the user has access
         accessible_projects = self.get_projects(request, organization, project_ids={-1})
+        # projects where the user has project membership
+        projects = self.get_projects(request, organization)
+
+        # org admins can modify projects and not have direct project access
+        if not projects and not request.access.has_scope("project:write"):
+            raise PermissionDenied
```

Comment:
Would this not fail if you were an org owner/manager and hit the endpoint with `org:admin`?

Reply:
do you mean with a token?

Reply:
Sorry I just realized how this works. You want Team Admins and Org Admins to be able to access this endpoint, so you set `org:read` so they can access it. `self.get_projects` returns something for team admins so that's covered. I don't quite understand what `if not projects and not request.access.has_scope("project:write"):` does.

Also, would this fail for an org member who has `org:read`?

Reply:
how does it fail for regular members with only `org:read`. Does `self.get_projects` always return `[]` for a member, and then it fails on the check. Also can you include a test to make sure it fails for a regular member

Reply:
there is an existing test for when you don't have access to the project, is this what you mean? the default role created in this test is member https://github.com/getsentry/sentry/blob/master/tests/sentry/api/endpoints/notifications/test_notification_actions_index.py#L216

Reply:
`self.get_projects` returns whatever projects the user has access to via being on the same team as the project. if we pass `include_all_accessible=True` into the function then we get all projects that the user can see.

https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/organization.py#L334-L338

Reply:
oh i think i see what you mean. it should fail for members even if they have access

Reply:
Oh yes that's the test case I meant